practicals/2-od.Rmd

```diff
@@ -157,117 +208,83 @@ ggplot(od_data_new) +
 ```
 
+Now that we have geometry attached to our data we can calculate other variables of interest. For example let's calculate the distacne travelled and see if it relates to the percentage of people who use active travel.
 
 ```{r}
-# transform
-od_central = od %>%
-  filter(o %in% zones_central$geo_code) %>% 
-  filter(d %in% zones_central$geo_code) 
-nrow(od_central) / nrow(od)
-desire_lines = od2line(od_central, zones_central)
 desire_lines$distance_direct_m = as.numeric(st_length(desire_lines))
-desire_lines = desire_lines %>% 
-  mutate(proportion_active = (bicycle + foot) / all)
 ```
 
-```{r, fig.show='hold', out.width="40%"}
-# visualise
-ggplot(desire_lines) +
-  geom_point(aes(distance_direct_m, proportion_active))
-ggplot(desire_lines) +
-  geom_point(aes(distance_direct_m, proportion_active, size = all), alpha = 0.3)
-```
+Note the use of `as.numeric` by default `st_length` and many other functions return a special type of result with `unit`. Here we force the results back into the basic R numerical value. But be careful! The units you get back depend on the coordinate reference system, so check your data before you assume what values mean.
 
-```{r}
-# model/visualise
-m1 = lm(proportion_active ~ 
-          distance_direct_m + I(distance_direct_m^2),
-        data = desire_lines)
-desire_lines = desire_lines %>% 
-  mutate(
-    new_active_travel = m1$fitted.values * car_driver,
-    new_total_active = new_active_travel + bicycle + foot,
-    new_proportion_active = new_total_active / all
-    ) %>% 
-  arrange(proportion_active)
+```{r, warning=FALSE, message=FALSE}
 ggplot(desire_lines) +
-  geom_point(aes(distance_direct_m, proportion_active, size = all), alpha = 0.3) +
-  geom_point(aes(distance_direct_m, new_proportion_active, size = all), alpha = 0.3, colour = "blue")
-
+  geom_point(aes(x = distance_direct_m, y = Active, size = all)) +
+  geom_smooth(aes(x = distance_direct_m, y = Active))
 ```
 
-```{r}
-# visualise
-ggplot(desire_lines) +
-  geom_sf(aes(colour = new_proportion_active, alpha = all))
-```
+The blue line is a smoothed average of the data. It shows a common concept in transport research, the distance decay curve. In this case it shows that the longer the journey the less likely people are to use active travel. But this concept applies to all kinds of travel decisions. For example you are more likely to travel to a nearby coffee shop than a far away coffee shop. Different types of trip have different curves, but most people always have a bias for shorter trips.
 
-```{r}
-library(tmap)
-tm_shape(desire_lines) +
-  tm_lines(palette = "-viridis", breaks = c(0, 5, 10, 20, 40, 100) / 100,
-    lwd = "all",
-    scale = 9,
-    title.lwd = "Number of trips",
-    alpha = 0.6,
-    col = c("proportion_active", "new_proportion_active"),
-    title = "Active travel (%)"
-  ) +
-  tm_scale_bar()
 
-```
+# Homework
+
+1. Read Chapters 2-5, 7, and 8 of [Geocomputation with R](https://r.geocompx.org/transport.html) 
```

Comment:
That's quite a lot to get through, I suggest making 7 and 8 bonus questions.

I'd rather everyone did the basics than asking too much and nobody completes the homework.

Reply:
I've made more bonus work